CSharp/BatchExplorer/Models/TaskModel.cs

```diff
@@ -97,7 +97,15 @@ public class TaskModel : ModelBase
         [ChangeTracked(ModelRefreshType.Children)]
         public bool HasOutputFiles
         {
-            get { return (this.OutputFiles != null && this.OutputFiles.Any()); }
+            get
+            {
+                if (this.OutputFiles == null || !this.OutputFiles.Any())
+                {
+                    this.RefreshAsync(ModelRefreshType.Children);
```

Comment:
Also I thought there was a reason we didn't do this which is that tasks which ran on Nodes which ahve since been deallocated will fail this every single time, popping up an annoying exception box each time.  I think I'd rather force a manual refresh than that.